src/gate/parserGnet/parserGnet.cpp

```diff
@@ -0,0 +1,436 @@
+/**
+ * \brief Implements translation from Liberty format file to GNet.
+ * \author <a href="mailto:anushakov@edu.hse.ru">Aleksander Ushakov</a>
+ */
+
+#include "parserGnet.h"
+#include "gate/simulator/simulator.h"
+
+#include <deque>
+#include <string>
+
+namespace GModel = eda::gate::model;
+namespace YLib = Yosys::hashlib;
+namespace RTlil = Yosys::RTLIL;
+
+using eda::gate::model::Gate;
+using eda::gate::model::GateSymbol;
+using GateId = eda::gate::model::Gate::Id;
+using eda::gate::simulator::Simulator;
+
+class YosysManager {
+public:
+  YosysManager() {
+    Yosys::yosys_setup();
+  }
+
+  ~YosysManager() {
+    Yosys::yosys_shutdown();
+  }
+};
+
+/// The functions creates inputs container.
+static void fillInputs(
+    const YLib::dict<RTlil::IdString, RTlil::Wire*> &wires,
+    GModel::GNet &net,
+    std::map<size_t, GateId> &inputs) {
+  std::deque<int> tmp;
+  for (auto [name, wire]: wires) {
+    size_t index = name.index_;
+    if (wire->port_input) {
+      tmp.push_front(index);
+    }
+  }
+  for (auto index : tmp) {
+    GateId inputId = net.addIn();
+    inputs.emplace(index, inputId);
+  }
```

Comment:
`std::map` is a *sorted* container, that means iterations are in (some) order no matter what the insertion order is. Just use `inputs.emplace(index, inputId)` without an intermediate deque.